target/arm/sve_helper.c

```diff
@@ -866,6 +866,26 @@ DO_ZPZZ_PAIR(sve2_sminp_zpzz_h, int16_t, H1_2, DO_MIN)
 DO_ZPZZ_PAIR(sve2_sminp_zpzz_s, int32_t, H1_4, DO_MIN)
 DO_ZPZZ_PAIR_D(sve2_sminp_zpzz_d, int64_t, DO_MIN)
 
+DO_ZPZZ_PAIR(sve2_faddp_zpzz_h, float16, H1_2, DO_ADD);
```

Comment:
DO_ADD is an integer addition.  These need to use floating point operations.
Compare DO_ZPZZ_FP and sve_fadd_[hsd].
You will have to add the "void *status" operand to the helper, which means you can't directly use DO_ZPZZ_PAIR, but will need a new DO_ZPZZ_PAIR_FP.